streams/src/test/java/org/apache/kafka/streams/kstream/TimeWindowsTest.java

```diff
@@ -103,6 +104,11 @@ public void gracePeriodShouldEnforceBoundaries() {
         }
     }
 
+    @Test
+    public void oldAPIShouldSetDefaultGracePeriod() {
+        assertEquals(DEPRECATED_OLD_24_HR_GRACE_PERIOD - 3L, TimeWindows.of(ofMillis(3L)).gracePeriodMs());;
+    }
```

Comment:
I think, we need additional tests:

```suggestion
    @Test
    public void deprecatedDefaultGracePeriodShouldBeOneDay() {
        assertThat(DEPRECATED_OLD_24_HR_GRACE_PERIOD, is(Duration.ofDays(1).toMillis()));
    }

    @Test
    public void oldAPIShouldSetGracePeriodToDefaultGracePeriodMinusWindowSize() {
        final Duration windowSize = Duration.ofMillis(3L);
        assertThat(TimeWindows.of(windowSize).gracePeriodMs(), is(DEPRECATED_OLD_24_HR_GRACE_PERIOD - windowSize.toMillis()));
    }
    
    @Test
    public void oldAPIShouldSetGracePeriodToZeroWhenWindowSizeIsDefaultGracePeriod()() {
        final Duration windowSize = Duration.ofMillis(DEPRECATED_OLD_24_HR_GRACE_PERIOD);
        assertThat(TimeWindows.of(windowSize).gracePeriodMs(), is(0L));
    }    

    @Test
    public void oldAPIShouldSetGracePeriodToZeroWhenWindowSizeGreaterDefaultGracePeriod()() {
        final Duration windowSize = Duration.ofMillis(DEPRECATED_OLD_24_HR_GRACE_PERIOD + 1);
        assertThat(TimeWindows.of(windowSize).gracePeriodMs(), is(0L));
    }    

```
The first test makes the deprecated default grace period robust against refactorings.
The other three tests correspond to the three important cases, namely:
- window size less than default grace period
- window size equal to default grace period
- window size greater than default grace period 

You might need some imports for this code to work.

Similar needs to be done for `SessionWindows`.